inloop/solutions/views.py

```diff
@@ -99,15 +100,57 @@ def get(self, request, **kwargs):
 
         # TODO: PrettyPrinters should be configurable (for now, we only have one for JUnit)
         result = solution.testresult_set.last()
-        xml_reports = junit.checkeroutput_filter(result.testoutput_set)
-        testsuites = [
-            junit.xml_to_dict(xml) for xml in xml_reports
-        ]
+
+        xml_reports_junit = testing.checkeroutput_filter(result.testoutput_set)
+        testsuites = [testing.xml_to_dict(xml) for xml in xml_reports_junit]
+
+        parser = Parser(solution=solution)
+        junit_context = parser.context(
+            startswith="TEST-",
+            endswith=".xml",
+            filter_keys=["testcase", "system-out", "system-err", "failures", "errors", "failure", "error"]
+        )
+        checkstyle_context = parser.context(
+            startswith="checkstyle_errors",
+            endswith=".xml",
+            filter_keys=[]
+        )
+        spotbugs_context = parser.context(
+            startswith="spotbugs_errors",
+            endswith=".xml",
+            filter_keys=[]
+        )
+        pmd_context = parser.context(
+            startswith="pmd_errors",
+            endswith=".xml",
+            filter_keys=[]
+        )
+
+        checkstyle_files = Parser.extract(data=checkstyle_context["data"], key="file")
+
+        # Assign source code files
+        for file in checkstyle_files:
```

Comment:
This functionality is risky in terms of error handling. I will refactor it in the next step.